lib/libzfs/libzfs_pool.c

```diff
@@ -1251,16 +1227,6 @@ zpool_create(libzfs_handle_t *hdl, const char *pool, nvlist_t *nvroot,
 			goto create_failed;
 		}
 
-		if (nvlist_exists(zc_fsprops,
```

Comment:
I'm not understanding why this sanity check needs to be removed?  This check is part of the original design.

This is expected behavior:
```
$ sudo zpool create -O special_small_blocks=0 invalid xvdb
cannot create 'invalid': special_small_blocks property requires a special vdev
```

Reply:
Can you explain why during a `zpool create` you would need to set this property?  The default inheritable value is already zero.

Reply:
during zpool create I will set it to non-zero while there is no special vdev and then later add a special vdev.